### DISCUSSION TITLE
709

### DISCUSSION THREAD

**Discussion by giscus** [jojoldu/blog-comments#890] [Comments]
# 709

최근 여러 개발 리더분들을 뵐 기회가 있어서 이런 저런 이야기를 나눴다. 그러다보면 공통적으로 하는 이야기가 ChatGPT 로 인한 개발자의 자리에 대한 것이다. 어떤 분들은 그간 시니어들이 설계를 하고 구현은 주니어들이 하던 방식에서 ChatGPT를 활용해서 별도의 구현 코드 작성에 필요한 주니어들의 자리가 대체될 것 같다고 한다. 또 다른 의견으로는 이제 주니어들이 굳이 시니어에게 코드 리뷰나 더 좋은 코드에 대한 의견을 물어보기 보다는 ChatGPT에게 물어보면 될 것이고, 서버리스 환경까지 적극적으로 도입되다보면 시니어들의 자리가 굳이 필요할것인가 하는 것도 있었다. 대부분의 대중화된 프레임워크로 튜닝은 ChatGPT에게 물어보면 되고, 인프라나 미들웨어 등은 클라우드의 매니지드 서비스를 사용하면..

https://jojoldu.tistory.com/709

**Comment by soulduse** [comment]
정말 대격변에 시대에 살고 있는것 같아요, ChatGPT를 잘 활용하느냐 아니냐로 개발자의 능력이 나눠지진 않을까라는 생각도 문득 들더라구요, 위에서 말해주신 사례처럼 정말로 GPT를 사용해서 생산성이 증가했다라는 정량적인 측정이 가능하다면 개발자 채용 시장이 좀더 모수가 줄어들지 않을까라는 생각도 들구요. 참 여러모로 걱정도 되고 반대로 두근거리는 시기인것 같아요..!

**Reply by soulduse** [reply]
최근 다른 개발자분들과 ChatGPT 관련 이야기 해본적 있는데
이미 많은 부분에서 GPT를 활용하여 업무에 활용하고 있더라구요
```
- 테스트 코드 작성
- 비즈니스 로직 구현
- SQL 작성
- GPT와 페어프로그래밍
- 코드 최적화 작업
- 변수, 클래스 네이밍 결정
```
정말 많은 생각이 드는 요즘 입니다.

**Reply by jojoldu** [reply]
ㅋㅋ 완전 동의해 현우야 ㅋㅋㅋ 현우 같이 풀스택 개발자들은 이제 진짜 날개를 펼것같다는 생각하고 들고 ㅋㅋ
나도 최근에 새롭게 배우는 언어나 생태계를 GPT를 통해 한번 진행해보는 실험을 해볼까 하는데,
한번 그 결과를 후기로 남겨볼까해 ㅋㅋ

**Comment by Voyager003** [comment]
gpt가 알고리즘을 풀어주는 시대가 왔는데, 입사를 위한 관문이 코테인것도 참 아이러니하네요 
PCCP같은 시험이 공인 시험으로 선정되고, 이런 유형의 테스트가 늘어날 수도 있겠군요

**Reply by jojoldu** [reply]
저도 아마 온라인 코딩 테스트는 조금 애매할 수도 있겠다는 생각이 들었습니다 :) 
오프라인 코딩 테스트나 과제 기반 테스트가 좀 더 활성화되지 않을까 하는 생각입니다 :)

**Comment by abel9851** [comment]
안녕하세요. 2년차 백엔드 개발자입니다. 
향로님의 글을 읽고 막연히 느꼈던 불안감이 해소되었습니다.
특히 선언형 프로그래밍, VScode의 자동완성기능과 연관지어 설명하시는 부분을 읽고
앞으로 chatGPT를 어떻게 대해야할지에 대해 제 안에서 정리가 됬습니다.
정말 감사합니다.

블로그에 생각을 정리하는데 항로님의 이 글의 URL을 써도 될지 여쭙고 싶습니다.

**Reply by jojoldu** [reply]
아! 네네 물론입니다 :) URL 사용하셔도 됩니다 :)
